tools/build.cmd

```diff
@@ -157,6 +157,21 @@ rem ----------------------------------------------------------------------------
 rem this is setting the cmake path in a quoted way
 set "cmake-root=%build-root%\build"
 
+rem Figure out which CMake generator to use
+if "%VisualStudioVersion%" == "14.0" (
+    set "cmake-generator=Visual Studio 14 2015"
+) else if "%VisualStudioVersion%" == "15.0" (
+    set "cmake-generator=Visual Studio 15 2017"
+)
+if NOT DEFINED cmake-generator (
+    echo ERROR: VisualStudioVersion not defined, or not one of ^(14.0, 15.0^)
+    exit /b 1
+)
+if "%build-platform%" == "x64" (
+    set "cmake-generator=!cmake-generator! Win64"
```

Comment:
Didn't know batch files supported variable expansion using `!` symbols! Is the usual `%` syntax not permissible here?